src/renderer/js/page/subscriptions/view.jsx

```diff
@@ -0,0 +1,106 @@
+import type { Subscription } from "redux/reducers/subscriptions";
+
+type SavedSubscriptions = Array<Subscription>
+
+type Props = {
+  doFetchClaimsByChannel: (string, number) => any,
+  savedSubscriptions: SavedSubscriptions,
+  // TODO build out claim types
+  subscriptions: Array<any>,
+  setHasFetchedSubscriptions: () => void,
+  hasFetchedSubscriptions: boolean
+};
+
+export default class extends React.PureComponent<Props> {
+  // setHasFetchedSubscriptions is a terrible hack
+  // it allows the subscriptions to load correctly when refresing on the subscriptions page
+  // currently the page is rendered before the state is rehyrdated
+  // that causes this component to be rendered with zero savedSubscriptions
+  // we need to wait until persist/REHYDRATE has fired before rendering the page
+  componentDidMount() {
+    const { savedSubscriptions, setHasFetchedSubscriptions } = this.props;
+    if (savedSubscriptions.length) {
+      this.fetchSubscriptions(savedSubscriptions);
+      setHasFetchedSubscriptions();
+    }
+  }
+
+  componentWillReceiveProps(props: Props) {
+    const { savedSubscriptions, hasFetchedSubscriptions, setHasFetchedSubscriptions } = props;
+
+    if (!hasFetchedSubscriptions && savedSubscriptions.length) {
+      this.fetchSubscriptions(savedSubscriptions);
+      setHasFetchedSubscriptions();
+    }
+  }
+
+  fetchSubscriptions(savedSubscriptions: SavedSubscriptions) {
+    const { doFetchClaimsByChannel } = this.props;
+    if (savedSubscriptions.length) {
+      // can this use batchActions?
+      savedSubscriptions.forEach(sub => {
+        doFetchClaimsByChannel(sub.uri, 1);
+      });
+    }
+  }
+
+  render() {
+    const { subscriptions, savedSubscriptions } = this.props;
+
+    let someClaimsNotLoaded;
+    for (var i = 0; i < subscriptions.length; i++) {
+      const subscription = subscriptions[i];
+      if (!subscription.claims.length) {
+        someClaimsNotLoaded = true;
+        break;
+      }
+    }
+
+    const fetchingSubscriptions =
+      !!savedSubscriptions.length && (subscriptions.length !== savedSubscriptions.length || someClaimsNotLoaded);
+
+    return (
+      <main className="main main--no-margin">
+        <SubHeader modifier="full-width" />
+        {!savedSubscriptions.length && (
+          <div>{__("You haven't subscribed to any channels yet")}</div>
+        )}
+        {!!savedSubscriptions.length && (
+          <div>
+            {!!subscriptions.length &&
+              subscriptions.map(subscription => {
+                if (!subscription.claims.length) {
+                  // will need to update when you can subscribe to empty channels
+                  // for now this prevents issues with FeaturedCategory being rendered
+                  // before the names (claim uris) are populated
+                  return "";
+                }
+
+                // creating uris for each subscription file
+                const names = subscription.claims.slice().map(claim => {
```

Comment:
This could conceivably be moved into the/a selector. I get the impression that most data wrangling should happen inside of Redux rather than React.